core/src/test/java/org/apache/iceberg/TestOverwriteWithValidation.java

```diff
@@ -747,6 +747,60 @@ public void testConcurrentConflictingPositionDeletesOverwriteByFilter() {
         overwrite::commit);
   }
 
+  @Test
+  public void testConcurrentConflictingDataFileDeleteOverwriteByFilter() {
```

Comment:
This test would previously fail as the commit would succeed.

Reply:
The error here is because file 2 is both modified and deleted correct?

Reply:
Yeah, we concurrently removed data from a partition what we are trying to overwrite.